### PR TITLE
Update using generic

### PR DESCRIPTION
**Pull request by otaviojava**
# Changes

It is pretty similar to the https://github.com/jakartaee/data/pull/497 for getting consistency
If we say that it is a safer approach, the majority of the sample in the spec should follow it.

### CODE REVIEW

**Review by gavinking — APPROVED** [review]

File: spec/src/main/asciidoc/repository.asciidoc [hunk 1]
```diff
@@ -1148,7 +1148,7 @@ You can combine sorting with a starting page and maximum page size by using prop
 
 [source,java]
 ----
-PageRequest<?> pageRequest = PageRequest.ofSize(20).page(1).sortBy(Sort.desc("price"));
+PageRequest<Product> pageRequest = PageRequest.ofSize(20).page(1).sortBy(Sort.desc("price"));
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:1151]:
```suggestion
PageRequest<Product> pageRequest = PageRequest.of(Product.class).size(20).page(1).sortBy(Sort.desc("price"));
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 2]
```diff
@@ -1339,15 +1339,15 @@ For example,
...
+                                 int zipcode, PageRequest<Customer> pageRequest);
 }
 ----
 
 You can obtain the initial page relative to an offset and subsequent pages relative to the last entity of the current page as follows,
 
 [source,java]
 ----
-for (PageRequest<?> p = PageRequest.ofSize(50); p != null; ) {
+for (PageRequest<Customer> p = PageRequest.ofSize(50); p != null; ) {
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:1350]:
```suggestion
for (PageRequest<Customer> p = PageRequest.of(Customer.class).size(50); p != null; ) {
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 3]
```diff
@@ -1359,7 +1359,7 @@ Or you can obtain the next (or previous) page relative to a known entity,
 [source,java]
 ----
 Customer c = ...
-PageRequest<?> p = PageRequest.ofSize(50).afterKeyset(c.lastName, c.firstName, c.id);
+PageRequest<Customer> p = PageRequest.ofSize(50).afterKeyset(c.lastName, c.firstName, c.id);
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:1362]:
```suggestion
PageRequest<Customer> p = PageRequest.of(Customer.class).size(50).afterKeyset(c.lastName, c.firstName, c.id);
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 4]
```diff
@@ -1417,15 +1417,15 @@ Here is an example where an application uses `@Query` to provide a partial query
...
 ----
 
 Example traversal of pages:
 
 [source,java]
 ----
-for (PageRequest<?> p = PageRequest.ofSize(25).sortBy(Sort.desc("yearBorn"),
+for (PageRequest<Customer> p = PageRequest.ofSize(25).sortBy(Sort.desc("yearBorn"),
                                                       Sort.asc("name"),
                                                       Sort.asc("id"));
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:1430]:
```suggestion
for (PageRequest<Customer> p = Order.by(_Customer.yearBorn.desc(),
                                        _Customer.name.asc(),
                                        _Customer.id.asc())
                                    .pageSize(25);
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 5]
```diff
@@ -1452,7 +1452,7 @@ Obtaining the next 10 products that cost $50.00 or more:
 [source,java]
 ----
 float priceMidpoint = 50.0f;
-PageRequest<?> pageRequest = PageRequest.ofSize(10)
+PageRequest<Product> pageRequest = PageRequest.ofSize(10)
                                         .sortBy(Sort.asc("price"), Sort.asc("id"))
                                         .afterKeyset(priceMidpoint, 0L);
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:1457]:
```suggestion
PageRequest<Product> pageRequest = Order.by(_Product.price.asc(), _Product.id.asc())
                                        .pageSize(10)
                                        .afterKeyset(priceMidpoint, 0L);
```

**Comment by otaviojava** [comment]
> It's fine to change all of these examples, but we should do so in a way that doesn't generate warnings, meaning we need to pair that either with using the correctly typed Sort from the metamodel or else create the PageRequest with the correct type in the first place. I added suggestions to do all of that.

Thank you, Nathan.
That was precisely the goal, but I made this mistake. 
Thank you for fixing it.

**Pull request merged**